### PR TITLE
Simplify disabled styles

### LINKED ISSUE #27784: Bootstrap 4.1.2 `.close` hover focus styles are too specific
https://github.com/twbs/bootstrap/commit/4f1bb98587411d6db16783ea116ff6d5d8f8fa49 from 4.1.2 moved hover focus styles on the `.close` component into `&:not(:disabled):not(.disabled)` making it too specific. This breaks hover focus states for those that extended the close component from previous versions.

https://github.com/twbs/bootstrap/commit/eb5d0b49dc7439fa1cc3d0c41406eef00de149c1 makes it a little better, but the problem is still there.

### PR DESCRIPTION
**Pull request by MartijnCuppens**
Since we can now officially use `pointer-events` since IE10 is dropped, we can simplify our code a bit and decrease our selector specificity of all active `.btn`s and `.close` button.

I've also removed the `cursor: auto;` from the links since `pointer-events: none` is applied. And I 've added `pointer-events: none` to the form ranges which doesn't require to reset the cursor styles.

Closes https://github.com/twbs/bootstrap/pull/27785
Fixes #27784

TODO:
- [x] Document the effect of `pointer-events`

### CODE REVIEW

**Review by mdo — COMMENTED** [review]
Oh hell yeah, love this. Let’s document it in the migration guidelines, and possibly the docs for each documented example featuring a disabled component.

**Comment by ysds** [comment]
I was thinking about this too 👍 

For compatibility, disabled should be written after active.

Before:
image

https://codepen.io/fellows3/pen/YzKQLZo

Current:
image

https://codepen.io/fellows3/pen/ExYXLgR

File: scss/_close.scss [hunk 1]
```diff
@@ -12,11 +12,14 @@
...
 
-  &:not(:disabled):not(.disabled) {
-    &:hover,
-    &:focus {
-      opacity: .75;
-    }
+  &.disabled,
+  &:disabled {
+    pointer-events: none;
+  }
```
Comment (ysds) [inline scss/_close.scss]: It is more readable that this block is written after the `: hover,: focus`.

File: scss/_buttons.scss [hunk 2]
```diff
@@ -32,12 +30,13 @@
   // Disabled comes first so active can properly restyle
   &.disabled,
   &:disabled {
+    pointer-events: none;
     opacity: $btn-disabled-opacity;
     @include box-shadow(none);
   }
```
Comment (ysds) [inline scss/_buttons.scss]: This block should also move after the `.active`.

**Comment by MartijnCuppens** [comment]
I've tackled @ysds' comments. The colors have been updated in the meantime, that's why there's still a difference. The updated codepen of the master branch is https://codepen.io/MartijnCuppens/pen/povEOBB

**Comment by XhmikosR** [comment]
Is there something else preventing us from merging this?

**Review by ysds — APPROVED** [review]
LGTM 👍

**Comment by XhmikosR** [comment]
@patrickhlauke LGTY?

**Review by patrickhlauke — APPROVED** [review]

**Pull request merged**